File: Python/p4.py
n = int(input("Введите длину массива: "))
X = [0] * n

for i in range(len(X)):
    X[i] = float(input(f"x({i}) = "))

A = float(input("Введите A: "))

index = -1
for i in range(len(X)):
    if X[i] > A:
        index = i
        break

if index != -1:
    print(f"Первый элемент массива, значение которого больше A x({index}): {X[index]}")
else:
    print("В массиве нет элементов, значение которого больше A")

for i in range(4, len(X), 4):
    X.pop(i)

if len(X) > 0:
    sum_ = 0
    count = 0
    for i in range(0, len(X), 2):
        if i % 2 == 0:
            sum_ += X[i]
            count += 1
    if count > 0:
        mean = sum_ / count
        for i in range(min(n, len(X)) - 1, max(0, len(X) - 3) - 1, -1):
            X[i] = mean

print("Измененный массив: ")
for i in range(len(X)):
    print(f"x({i}) = {X[i]}")

File: Java/Practice/p4.java
package Practice;

import java.util.Arrays;
import java.util.Scanner;

public class p4 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Введите длину массива: ");
        int n = scanner.nextInt();
        System.out.print("Введите A: ");
        double A = scanner.nextDouble();

        double[] X = new double[n];

        for (int i = 0; i < n; i++) {
            System.out.print("x(" + i + ") = ");
            X[i] = scanner.nextDouble();
        }


        int index = -1;
        for (int i = 0; i < X.length; i++) {
            if (X[i] > A) {
                index = i;
                break;
            }
        }

        if (index != -1) {
            System.out.println("Первый элемент массива, значение которого больше, A x(" + index +
                    "): " + X[index]);
        } else {
            System.out.println("В массиве нет элементов, значение которого больше A");
        }
        int countindex = 0;
        for (int i = 0; i < X.length; i++) {
            if ((i + 1) % 5 != 0) {
                X[countindex] = X[i];
                countindex++;
            }
        }
        X = Arrays.copyOf(X, countindex);


        double sum = 0;
        int count = 0;
        for (int i = 0; i < X.length; i += 2) {
            sum += X[i];
            count++;
        }

        if (count > 0) {
            double mean = sum / count;
            for (int i = X.length - 1; i >= X.length - 3; i--) {
                X[i] = mean;
            }
        }

        System.out.println("Измененный массив: ");
        for (int i = 0; i < X.length; i++) {
            System.out.println("x(" + i + ") = " + X[i]);
        }
    }
}

